halo2_proofs/src/plonk/prover.rs

```diff
@@ -635,7 +482,7 @@ pub fn create_proof<
     // Evaluate the permutations, if any, at omega^i x.
     let permutations: Vec<permutation::prover::Evaluated<C>> = permutations
         .into_iter()
-        .map(|permutation| -> Result<_, _> { permutation.evaluate(pk, x, transcript) })
+        .map(|permutation| -> Result<_, _> { permutation.construct().evaluate(pk, x, transcript) })
```

Comment:
I was wondering if there was a way to parallelize here. But seems that the transcript will now allow it.

Reply:
I think most time is spent in `eval_polynomial`, which has now been parallelized. Because I currently have just optimized using the EVM circuit, and the EVM circuit does not have these permutations, I think there's a very large chance that further improvements are possible related to those though for other circuits!